esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -1431,22 +1505,31 @@ void EsdlServiceImpl::prepareFinalRequest(IEspContext &context,
 
     // Process Custom Request Transforms
 
-    if (m_serviceLevelCrtFail)
-        throw MakeStringException(-1, "%s::%s disabled due to service-level Custom Transform errors. Review transform template in configuration.", srvdef.queryName(), mthName);
+    if (m_serviceScriptError.length())
+        throw MakeStringException(-1, "%s::%s disabled due to service-level Custom Transform errors. [%s]. Review transform template in configuration.", srvdef.queryName(), mthName, m_serviceScriptError.str());
 
-    StringAttr *crtErrorMessage = m_methodCRTransformErrors.getValue(mthName);
+    StringAttr *crtErrorMessage = m_methodScriptErrors.getValue(mthName);
     if (crtErrorMessage && !crtErrorMessage->isEmpty())
-        throw MakeStringException(-1, "%s::%s disabled due to method-level Custom Transform errors: %s. Review transform template in configuration.", srvdef.queryName(), mthName, crtErrorMessage->str());
+    {
+        VStringBuffer msg("%s::%s disabled due to ESDL Script error(s). [%s]. Review transform template in configuration.", srvdef.queryName(), mthName, crtErrorMessage->str());
+        throw makeWsException( ERR_ESDL_BINDING_INTERNERR, WSERR_CLIENT, "ESDL", "%s", msg.str());
+    }
 
     if (scriptContext)
     {
-        IEsdlCustomTransform *methodCrt = m_customRequestTransformMap.getValue(mthName);
+        IEsdlTransformSet *serviceCRTs = m_transforms->queryMethodEntryPoint("", ESDLScriptEntryPoint_Legacy);
+        IEsdlTransformSet *methodCRTs = m_transforms->queryMethodEntryPoint(mthName, ESDLScriptEntryPoint_Legacy);
+
+        IEsdlTransformSet *serviceRTs = m_transforms->queryMethodEntryPoint("", ESDLScriptEntryPoint_BackendRequest);
+        IEsdlTransformSet *methodRTs = m_transforms->queryMethodEntryPoint(mthName, ESDLScriptEntryPoint_BackendRequest);
```

Comment:
These aren't unordered and ordered.  These are entry point names.  They are equavent to each other and if I add a "BackEndRequest" entry point to a service that already has Legacy entry points it will work.  Whether things are ordered or now is based on how the xml is added as CDATA or inline.

Reply:
I get the technicalities, but in reality there is no existing legacy markup that uses CDATA (thus all such markup is unordered) and new markup will likely use CDATA (and be ordered). I question whether it is necessary and whether an arbitrary decision that legacy transforms will be processed first will be sufficient. Maybe I (as a potential end user) want new markup to set things up for the legacy, so should we support an option to switch the order? Maybe I have an unrealistic expectation that I should somehow be able to intermingle old and new.

My thought (as a reviewer) is that we should support both, but not together. It's just my opinion, and I'm probably still influenced by my abandoned refactoring where I wasn't even looking for legacy if new markup was found. I also wasn't allowing legacy syntax to include new operations. It was pick one or the other, but never both.

Reply:
We will eventually deprecate the old name, but I don't want to dictate that people have to do that right away just to use a new tag.  I'm going to fix it so that the order will be preserved across both types.